vm/src/stdlib/zlib.rs

```diff
@@ -111,15 +118,187 @@ fn zlib_decompress(
     }
 }
 
-fn zlib_error(message: &str, vm: &VirtualMachine) -> PyBaseExceptionRef {
-    let module = vm
-        .get_attribute(vm.sys_module.clone(), "modules")
-        .unwrap()
-        .get_item("zlib", vm)
-        .unwrap();
+fn zlib_decompressobj(
+    wbits: OptionalArg<u8>,
+    zdict: OptionalArg<PyBytesLike>,
+    vm: &VirtualMachine,
+) -> PyDecompress {
+    let wbits = wbits.unwrap_or(MAX_WBITS);
+    let mut decompress = Decompress::new_with_window_bits(true, wbits);
+    if let OptionalArg::Present(dict) = zdict {
+        dict.with_ref(|d| decompress.set_dictionary(d).unwrap());
+    }
+    PyDecompress {
+        decompress: Mutex::new(decompress),
+        eof: AtomicCell::new(false),
+        unused_data: Mutex::new(PyBytes::new(vec![]).into_ref(vm)),
+        unconsumed_tail: Mutex::new(PyBytes::new(vec![]).into_ref(vm)),
+    }
+}
+#[pyclass(name = "Decompress")]
+#[derive(Debug)]
+struct PyDecompress {
+    decompress: Mutex<Decompress>,
+    eof: AtomicCell<bool>,
+    unused_data: Mutex<PyBytesRef>,
```

Comment:
Why not using `Vec<u8>`?

Reply:
Oh, there's supposed to be a getter for it, so it's that when somebody does `decompressor.unused_data` multiple times it doesn't clone the data, just the reference.